client/src/Components/JournalEntry/JournalEntry.js

```diff
@@ -1,12 +1,48 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "react-router-dom";
 import { DateTime } from 'luxon';
 import './JournalEntry.css';
+import { updateEntry } from "../../utilities/api-calls";
 
-const JournalEntry = ({ journal }) => {
+const JournalEntry = ({ journal, entryID }) => {
 
-  const {date, feeling, activity, journal_entry} = journal;
-  const formattedDate = DateTime.fromFormat(date, 'yyyy-MM-dd')
+  const {date, feeling, activity, journal_entry } = journal;
+
+  const [entry, setEntry] = useState(journal_entry);
+  const [savedEntry, setSavedEntry] = useState(journal_entry);
+  const [edit, setEdit] = useState(false);
+  
+
+  const formattedDate = DateTime.fromFormat(date, 'yyyy-MM-dd');
+  
+  const changeEdit = () => {
+    setEdit(!edit);
+  }
+
+  const handleChange = (event) => setSavedEntry(event.target.value);
+
+  const cancelEdit = () => {
+    changeEdit();
+    setSavedEntry(entry);
+  }
+
+  const saveEntry = () => {
+    updateEntry(savedEntry, entryID);
+    setEntry(savedEntry);
+    changeEdit();
+  }
+
+  const showEntry = <p className='journal-text'>{entry}</p>
+  const entryButtons = <div>
+                          <Link to={'/how-you-felt'} className="uni-btn">←</Link>
+                          <button className="uni-btn" onClick={changeEdit}>update</button>
+                        </div>
+
+  const showForm = <textarea className='journal-content' value={savedEntry} onChange={handleChange}/>
+  const formButtons = <div>
+                        <button className="uni-btn" onClick={cancelEdit}>cancel</button>
+                        <button className="uni-btn" onClick={saveEntry}>save</button>
+                      </div>
```

Comment:
Not getting too in depth on this comment because we worked on it together, but awesome work here!! So stoked to implement this functionality!

Reply:
YALL this is amazing. This cleans up the return for this component so nicely. Super super great work.